tests/unit/raft_test.cpp

```diff
@@ -321,6 +321,23 @@ TEST_F(raft_test, test_raft_serializer_basic) {
     ASSERT_TRUE(ser.end_of_buffer());
 }
 
+TEST_F(raft_test, test_raft_serializer_read) {
+    auto new_log = nuraft::buffer::alloc(32);
+    auto test_hash = cbdc::hash_from_hex(
+        "cb7b43951ffcfe400a5432749a79096e632ef2e6328a28049c9af55b85fb260d");
+    auto ser = cbdc::nuraft_serializer(*new_log);
+    ser << test_hash;
+    ser.reset();
+    ASSERT_FALSE(ser.end_of_buffer());
+    auto read_output = nuraft::buffer::alloc(32);
+    ASSERT_TRUE(ser.read(read_output->data(), 32));
+    ASSERT_EQ(*read_output->data(), *test_hash.data());
+    ser.reset();
+    ser.advance_cursor(10);
+    ASSERT_TRUE(ser.read(read_output->data(), 22));
+    ASSERT_EQ(*read_output->data(), *(test_hash.data() + 10));
```

Comment:
So, I believe I understand what this is testing (skipping the first 10 bytes and reading the other 22, do those 22 bytes match what we expect), but I'm not quite sure why that should be tested. Lines 333 and 334 test that the full 32-bytes are equal.

Reply:
I suppose it is simply a test of the advance_cursor but if you like, i can simply remove it.

Reply:
I don't mind it, just wasn't clear on it. Maybe it's worth adding a comment above 336 to clarify that's the aim?

Reply:
Sounds good. I added a comment.